### DISCUSSION TITLE
Processing IQ data

### DISCUSSION THREAD

**Discussion by jaywai412** [mtex-toolbox/mtex#560] [Ask Anything] [answered]
Hi,

I succeeded to read iq data from osc files. 
But, there are a few undefined white data points in the plot.
I tried to fill the data using the following command as guided in the manual, but it did not work: ebsd_filled = fill(ebsd,grains);
Doesn't this function work for IQ data?

Another thing I am trying to do is to plot IQ values along a line crossing a particular grain boundary (like a height profile).
Is there an API that can sample data along a line specified with coordinates (ebsd.x and ebsd.y)?

Thank you.

**Comment by kilir** [comment, marked as answer]
Hi,
fill() does not interpolate anything unrealted to orientations.

For the selection of ebsd data along a line, seee  spatialProfile(), 
e.g.
```
mtexdata titanium
grains = ebsd.calcGrains;
plot(ebsd,ebsd.prop.ci)
hold on
plot(grains.boundary)
%xy = ginput(2)
xy= [100, 100; 800, 800]
[ebsd_l d] = spatialProfile(ebsd,xy)
hold on
plot(xy(:,1),xy(:,2),'-r')
hold off
figure
plot(d,ebsd_l.prop.ci)
```
To select the start/end points of the line manually, use ginput().

Cheers,
Rüdiger

**Reply by jaywai412** [reply]
Hi,
It works. 
Thank you very much for your help.
